components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -2485,135 +2485,140 @@ public static NativeArray jsFunction_getAllSubscriptions(Context cx,
 
             Subscriber subscriber = new Subscriber(username);
             APIConsumer apiConsumer = getAPIConsumer(thisObj);
-            Application[] applications = apiConsumer.getApplications(new Subscriber(username));
-            if (applications != null) {
+            //Application[] applications = apiConsumer.getApplications(new Subscriber(username));
+            //if (applications != null) {
+            Set<SubscribedAPI> subscribedAPIs = apiConsumer.getSubscribedAPIs(subscriber);
+            if (subscribedAPIs != null) {
                 int i = 0;
-                for (Application application : applications) {
-                    if (ApplicationStatus.APPLICATION_APPROVED.equals(application.getStatus())) {
-                        NativeObject appObj = new NativeObject();
-                        appObj.put("id", appObj, application.getId());
-                        appObj.put("name", appObj, application.getName());
-                        appObj.put("callbackUrl", appObj, application.getCallbackUrl());
-                        APIKey prodKey = getAppKey(application, APIConstants.API_KEY_TYPE_PRODUCTION);
-
-                        boolean prodEnableRegenarateOption = true;
-                        if (prodKey != null && prodKey.getAccessToken() != null) {
-                            appObj.put("prodKey", appObj, prodKey.getAccessToken());
-                            appObj.put("prodConsumerKey", appObj, prodKey.getConsumerKey());
-                            appObj.put("prodConsumerSecret", appObj, prodKey.getConsumerSecret());
-                            if (prodKey.getValidityPeriod() == Long.MAX_VALUE) {
-                                prodEnableRegenarateOption = false;
-                            }
-                            appObj.put("prodRegenarateOption", appObj, prodEnableRegenarateOption);
-                            appObj.put("prodAuthorizedDomains", appObj, prodKey.getAuthorizedDomains());
+             //   for (Application application : applications) {
+             //       if (ApplicationStatus.APPLICATION_APPROVED.equals(application.getStatus())) {
+             //           NativeObject appObj = new NativeObject();
+                for (SubscribedAPI subscribedAPI : subscribedAPIs) {
+                    Application application = subscribedAPI.getApplication();
+                    NativeObject appObj = new NativeObject();
+                    appObj.put("id", appObj, application.getId());
```

Comment:
Thanks for the comment. Constants added in the commit 79df88ba93e7ae485106a417158a645165096439